lib/xo.ts

```diff
@@ -251,6 +251,30 @@ export class Xo {
 		}
 	}
 
+	/**
+	 * Checks if any directly specified files are ignored and throws an error.
+	 * @private
+	 */
+	async checkForIgnoredDirectFiles(globs: string[]): Promise<void> {
+		const {flatOptions} = await resolveXoConfig(this.linterOptions);
```

Comment:
Hmm this will cause the resolveXoConfig function to get ran 2x I think.

Instead of calling this function maybe we can acheive the same thing by doing 

```ts
await this.setCoConfig();
this.xoConfig // instead of flatOptions
```